docs/compiling.rst

```diff
@@ -169,52 +172,120 @@ can refer to the same [cmake_example]_ repository for a full sample project
 
 .. _Config file: https://github.com/pybind/pybind11/blob/master/tools/pybind11Config.cmake.in
 
-Advanced: interface library target
-----------------------------------
+New: FindPython support
+-----------------------
+
+CMake 3.12+ (3.15+ recommended) added a new module called FindPython that had a
+highly improved search algorithm and modern targets and tools. If you use
+FindPython, pybind11 will detect this and use the existing targets instead:
+
+.. code-block:: cmake
+
+    cmake_minumum_required(VERSION 3.15...3.18)
+    project(example LANGUAGES CXX)
+
+    find_package(Python COMPONENTS Interpreter Development REQUIRED)
+    find_package(pybind11 CONFIG REQUIRED) # or add_subdirectory
```

Comment:
Doesn't `find_package` need that some search path is set up correctly? (so that it can find the pybind11 sub-directory)

Reply:
It's for install directories, not subdirectories; add_subdirectory is used in that case. I can add that here too if it helps.

Reply:
(Assuming that's what you meant. If you are looking for the installed config, it should often be found, though you can set several different variables, like `pybind11_DIR`, etc. Docs there will be expanded after working on the python packaging a bit, in the near future)

Reply:
Got it -- I've never used the install feature of pybind11 (all of my projects use git submodules), so I don't have a great intuition about those parts.

Reply:
> all of my projects use git submodules

+1!